File: Sing/DatabaseConnectionSingleton.java
// Clase que implementa el patrón Singleton
public class DatabaseConnection {
    // Única instancia de la clase
    private static DatabaseConnection instance;

    // Constructor privado para evitar instancias externas
    private DatabaseConnection() { }

    // Método estático para obtener la instancia única
    public static DatabaseConnection getInstance() {
        if (instance == null) {
            instance = new DatabaseConnection();
        }
        return instance;
    }

    // Método de ejemplo que representa una acción de la conexión
    public void connect() {
        System.out.println("Connecting to the database...");
    }
}

// Uso del patrón Singleton
public class Main {
    public static void main(String[] args) {
        DatabaseConnection connection = DatabaseConnection.getInstance();
        connection.connect();
    }
}

File: Sing/DatabaseConnectionSingleton.cs
// Clase que implementa el patrón Singleton
public sealed class DatabaseConnection {
    // Instancia única de la clase
    private static DatabaseConnection instance = null;
    private static readonly object padlock = new object();

    // Constructor privado para evitar instancias externas
    private DatabaseConnection() { }

    // Propiedad estática para obtener la instancia única
    public static DatabaseConnection Instance {
        get {
            lock (padlock) {
                if (instance == null) {
                    instance = new DatabaseConnection();
                }
                return instance;
            }
        }
    }

    // Método de ejemplo que representa una acción de la conexión
    public void Connect() {
        Console.WriteLine("Connecting to the database...");
    }
}

// Uso del patrón Singleton
class Program {
    static void Main(string[] args) {
        DatabaseConnection connection = DatabaseConnection.Instance;
        connection.Connect();
    }
}
